bakerytest/bakerytest.go

```diff
@@ -127,3 +130,131 @@ func (d *Discharger) PublicKeyForLocation(loc string) (*bakery.PublicKey, error)
 	}
 	return nil, bakery.ErrNotFound
 }
+
+type discharge struct {
+	cavId string
+	c     chan error
+}
+
+// LoginDischarger is a Discharger that always requires interraction to
+// complete the discharge.
+type LoginDischarger struct {
+	Discharger
+	Mux *http.ServeMux
+
+	// mu protects the following fields.
+	mu      sync.Mutex
+	waiting map[string]discharge
+	id      int
+}
+
+// NewLoginDischarger returns a new LoginDischarger which uses
+// loginHandler to handle the login interraction. The login is completed
```

Comment:
I think we should be more specific about what "handling the login interaction" actually means - for example it should perhaps mention that it is called on /login and (I presume) it's the responsibility of the handler to call FinishLogin.